### ISSUE TITLE
(WCAG) Missing attribute in the validation message role="alert"

### ISSUE THREAD

**Issue by amerimohammed** [bsgriggs/mendix-dropdown-date-picker#6]
For screen readers to detect the validation messages role="alert" must be one of the attributes as it is the case in the default Mendix validation messages.

**Comment by bsgriggs** [comment]
released in version 4.1.1

**Comment by amerimohammed** [comment]
Thanks a lot for updating validation message accessibility.
Can you also add aria-label="clear" to **Clear Button** when active maybe a text field or something to comply with WCAG guidelines for screen readers?

**Comment by bsgriggs** [comment]
You can use the Clear Button Tooltip. This sets the "title" which will get read if aria-label is not there.

**Comment by amerimohammed** [comment]
The title is being added to the **span** inside the **button** not to the button itself

**Comment by bsgriggs** [comment]
Hi, i've made the update in version 4.1.2

**Comment by amerimohammed** [comment]
Screen reader does not read the label, maybe it is a good idea to add aria-labelledby to all the select items or to group them in fieldset

**Comment by bsgriggs** [comment]
NVDA and the Windows screen reader are reading it correctly. 

Please make sure you've upgraded to version 4.1.2

![Image](https://github.com/user-attachments/assets/fd6a9ab6-ccae-43d9-aed6-938a53884b59)

**Comment by amerimohammed** [comment]
You are right about the button it was fixed and also can read **_Year_** **_Month_** **_Day_** combo box, but I noticed that it does not read the main title **_Local_** or in my case **_Birthday_**

**Comment by bsgriggs** [comment]
Made a change in v4.2.0 to prepend each dropdown's aria-label by the on-screen system label

Also added an explicit widget-wide [aria-label property](https://github.com/bsgriggs/mendix-dropdown-date-picker/releases/download/v4.2.0/mendix.DropdownDatePicker.mpk).

Screen readers should read each dropdown as "{On-Screen Label} {Aria-Label} {Dropdown Label}"

**Comment by amerimohammed** [comment]
Thanks a lot for this solution
